### DISCUSSION TITLE
Variant analysis and stats

### DISCUSSION THREAD

**Discussion by AGHz97** [ORTools4MTEX/ORTools#14] [Q&A] [answered]
Hi Frank, 
I am analyzing an austenoferritic stainless steel EBSD map, and I am intersted to analyze the Kurdjmov-Sachs (K-S) variants between the prior ferritic grains and the austenic child grains. I plot the variant map below :
![variant](https://user-images.githubusercontent.com/108796260/177529170-5cec0339-6a3f-4b03-ab0d-36d19b7bb625.png)
I want to get some statistics about those variants in the whole EBSD data : their distributions, their weighted area ?
I found that we can do that for a selected grain, but my request concerne all the grains at the same time.

Thank you very much for your help,

Ali

**Comment by frankNiessen** [comment]
Hi Ali,

my apologies for the late reply but I’m currently on leave. 

If you check out the code of [Example 7](https://github.com/ORTools4MTEX/ORTools/blob/develop/ORTools_example7.m) from line 157 and forward you will see that all the variant analysis features can be applied to individual prior Austenite grains as well as entire EBSD/grain datasets. I think this is what you are looking for? 

Best wishes
Frank

**Comment by AGHz97** [comment, marked as answer]
Hi Frank,
Thank you very much for your answers.
I have already seen example number 7.
However, my question concerns the calculation of austenitic laths' area having the same orientation (or belonging to the same Kurdjmov-Sachs variant). As you can see in the figure below, for a single grain there are austenitic laths having the same orientation but which are cutting by other laths. What I am asking for is if there is a way to calculate the equivalent surface of these laths even if they are separated. 
For example, in the diagram below, I wonder if we can calculate the surface of the blue laths (surrounded by a red line) ?
![scheme](https://user-images.githubusercontent.com/108796260/182048839-0f278e66-29a3-47d9-ae84-4485fd22427e.png)

Thank you very much for your help,

**Reply by frankNiessen** [reply]
Hi again Ali, 

yes, this is already implemented in the grain click command, in example 7 this is showcased on lines 187 - 188:

```
%% Display variant information for a PAG by interactively clicking on a gamma grain
grainClick(job,'noScalebar','noFrame'); %Based on EBSD data
```
This command lets you choose any parent grain by clicking on it. Figure **Histogram: Relative frequency variant Id's** gives you the weighted area fraction of the variants, or in other words the area of each variant assuming an overall area of 1 for the parent grain.
![image](https://user-images.githubusercontent.com/41304940/184087342-a9a6fc5f-08c5-4a46-b2c2-73db86f055bd.png)

The command also returns a table with the values into MATLAB's command window from which you can copy the data directly. 

```
  variantId      Freq   
    _________    _________

        1         0.035613
        2          0.01349
        3         0.016727
        4         0.035748
        5         0.047484
        6         0.019965
        7         0.027789
        8         0.031161
        9        0.0087684
       10         0.017941
       11         0.040065
       12        0.0035074
       13         0.077432
       14         0.023068
       15          0.11372
       16          0.14178
       17         0.012276
       18         0.045596
       19          0.13841
       20         0.016323
       21         0.022663
       22         0.048968
       23         0.014569
       24         0.046945
```

I hope this is what you were looking for. 

Best wishes
Frank

**Reply by AGHz97** [reply]
Hi Frank,
Thank you very much for your answers ! 
Actually I have seen the grainClick function before. It does the job for just one primary grain. Is there a way to do it for all primary grains. I don't want it to show me a lot of histograms, just a single histogram representing the area distribution of each variant (**not the weighted area**) in the whole map.
As a final result, **I have to find the surface distribution of the variants in the whole EBSD map**.

Thank you for your help,

**Reply by frankNiessen** [reply]
If I understand you correctly you want to get the absolute area that is occupied by each variant in the entire map. You can do this by getting the area of each transformed grain.
```
areas = job.transformedGrains.area;
```
Then you get the variant Id's of each transformed grain.
```
vIds = job.variantId(job.isTransformed);
```
As a last step, you can loop over each variant and sum up the areas of the grains that are identified as the respective variants:
```
for ii = 1:24
      vArea(ii) = sum(areas(vIds == ii));
end
```

I guess there is a more pretty way without using a for loop to do this, but it will work. 
To get the relative percentage of each variant you can of course do: 

```
vArea / sum(vArea)*100
```

**Reply by AGHz97** [reply]
Hi again Frank,
Thanks for your detailed answers. the area of the variant in the whole EBSD map is intresting too. I have an other question : Is there any way to identify for each primary grain the Ids of the variants that belong to it ?  I can use that to answer my question. 
Thank you for your help,

**Reply by frankNiessen** [reply]
Sure, 

after the reconstruction you can plot the parent grains like this

`figure;
plot(job.parentGrains,job.parentGrains.meanOrientation,'linewidth',2)`

When you click on a grain, its grain ID will be printed into the MATLAB command window. 
Save this grain ID in a variable by executing:

`grainId = 546;`

You can then retrieve all the grains prior to reconstruction in this parent grain by executing:

`clusterGrains = job.grainsPrior(job.mergeId == grainId);`

Now make sure you are only looking at grains of the child phase:

`cGrains = clusterGrains(job.csChild);`

And then retrieve their variant Id's:

`variantIds = cGrains.variantId;`

Now you can work with this data, or just plot it:

```
figure; 
plot(cGrains,variantIds);
mtexColorbar
```

Best wishes
Frank

**Reply by AGHz97** [reply]
Thank you Frank, that's exactly what I am looking for. However, is there any way (or variables) to get the Ids of the plotted prior grain map without clicking on the grains ?

Thanks again

**Reply by frankNiessen** [reply]
Sorry, I just saw your question now. You can get the ID's of all parent grains by executing 
`job.grainsPrior.id`, but in this case you will not exactly know which of these grains is your parent grain.
